### ISSUE TITLE
Determine browser compatibility goals (requirements? indicators?)

### ISSUE THREAD

**Issue by patcon** [GCTC-NTGC/gc-digital-talent#2361]
<sup>_This initial comment is collaborative and open to modification by all._</sup>
## Task Summary

🎟️ **Re-ticketed from:** https://github.com/GCTC-NTGC/gc-digital-talent/pull/2307#issuecomment-1079052167
✅ **[Proposed] Success Criteria:** Everyone who has experience/expertise has been heard and considered, and a file or document exists within our codebase.

Just creating a place to track this conversation and potential action items in the backlog outside slack. (happy to see this closed if/when we're done whatever needs discussing/doing)

- Tues, Mar 29: @tristan-orourke's original thread in Slack: https://talent-cloud.slack.com/archives/CB424V61J/p1648565972627749
- Thurs, Mar 31: @JoshBeveridge's Slack thread: https://talent-cloud.slack.com/archives/CB424V61J/p1648732161132539?thread_ts=1648732146.807549&cid=CB424V61J

## To Do

- [ ] ...

cc: @petertgiles

**Comment by gggrant** [comment]
Follow Web Experience Toolkit guidance.

**Comment by petertgiles** [comment]
React 18 drops support for IE 11.  https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#dropping-support-for-internet-explorer

**Comment by patcon** [comment]
Some random aggregation of related stuff:

- upgrading to React 18 implicit involves giving up support of IE11
  - related: https://github.com/GCTC-NTGC/gc-digital-talent/issues/2404
- GoC maintains a list of deprecated technologies
  - Canada.ca: [Deprecated Government of Canada technologies](https://www.canada.ca/en/government/system/digital-government/policies-standards/enterprise-it-service-common-configurations/deprecated-technologies.html)
  - as of 2017, anything before IE11 is retired
- as of June 15 2022, IE11 on Windows 10 (the last one to support it) has just been retired (h/t Josh [on Slack](https://talent-cloud.slack.com/archives/C0259G6KXJ6/p1655125542050299))
  - > the Internet **Explorer 11 desktop** application will be retired and go **out of support on June 15, 2022**, for certain versions of Windows 10. ["certain version" exemptions means "server" and "embedded systems" OS'es, like traffic lights, not normal enterprise]
  - 🤞 so GoC is perhaps going to have to retire IE11 soon
- Web Experience Toolkit still supports IE11: https://wet-boew.github.io/wet-boew-documentation/decision/2.html
- CRA is one of our important stakeholders
  - Culturally, they have some of the more conservative [browser support commitments] to IE9 (https://www.canada.ca/en/revenue-agency/services/e-services/e-services-businesses/corporation-internet-filing/your-browser.html) (to be fair, this page is for external corporate users, not internal gov)
  - financial software is amongst the slowest to upgrade, which holds back all underlying systems
  - CRA is 5th lowest score for blocked tech services: https://isthisblockedinmydepartment.ca/ (proxy for likelihood of being on old systems?) h/t gray

## Potential Options (and Questions)
1. Go with the **official GoC policy on supported tech**: no upgrades that lose IE11 support until it shows up on the deprecation page: https://www.canada.ca/en/government/system/digital-government/policies-standards/enterprise-it-service-common-configurations/deprecated-technologies.html
    - benefit: simple. makes any software upgrade decisions straight-fwd. minimal risk.
    - risks: getting behind on software. not being able to take advantage of upstream bugfixes.
3. Wait for **user analytics (browser usage data)** to start flowing. Once we launch (and maybe make it through a full cycle of internal users), we can have a better idea of who's using our software and what they need.
    1. Q: Do we have a way to collect browser usage data of visitors/users? Are we allowed to do this?
    2. Q: Is it likely that [admin] users who present themselves in the system early in the hiring cycle will be the exact same 
ones who show up later? How long do we need to wait to know that our browser data is representative? (e.g., maybe some pool manager users only log in from work during the later phase, and for some reason have more restrictive workstations)
    - risks: not sure if we're able to collect browser data. might be bureaucratic. bad data hygiene habit.
4. **Ask appropriate CRA staff or manager** (who will oversee admin/manager users) what sort of systems those users might be on.
    - risks: might be wrong. might not want to offer a guess without slow research.
    - benefit: getting advice from on-the-ground. most direct.

**Comment by mnigh** [comment]
> GoC maintains a list of deprecated technologies
> Canada.ca: [Deprecated Government of Canada technologies](https://www.canada.ca/en/government/system/digital-government/policies-standards/enterprise-it-service-common-configurations/deprecated-technologies.html)

Any idea who actually maintains this? If so, can we send in a suggestion to add IE 11 to this list? I reported a problem with the information 😉, maybe that will get it added.

> Web Experience Toolkit still supports IE11: https://wet-boew.github.io/wet-boew-documentation/decision/2.html

Should we be following decisions on browser support from over four year ago?

> CRA is 5th lowest score for blocked tech services: https://isthisblockedinmydepartment.ca/ (proxy for likelihood of being on old systems?) h/t gray

_isthisblockedinmydepartment.ca_ is not always up to date or accurate and not an official source to make a decision base upon, but someone has reported that chrome was not blocked by cra:
![Screen Shot 2022-06-21 at 16 00 13](https://user-images.githubusercontent.com/3046459/174889464-bf4eb059-cc42-49f7-a9bb-f8926d162fd3.png)

**Comment by patcon** [comment]
> Any idea who actually maintains this?

Going up a few levels, I'm seeing "From: Treasury Board of Canada Secretariat" and "Shared Services Canada" up at the ["Digital Government" landing page](https://www.canada.ca/en/government/system/digital-government.html)

> Should we be following decisions on browser support from over four year ago?

My prior tech sensibilities say "no", but my policy sensibilities (which seem more valid here) don't really have a time dimension 🙃

**Comment by patcon** [comment]
omg, I'm going to close this. I got turned around thinking it was about old browser support. As far as I can tell, there's nothing more to say here now that we have [`.browserslistrc` files](https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/frontend/admin/.browserslistrc) that define exactly what we want :/
